backend/arm64/emit.mlp

```diff
@@ -987,21 +981,139 @@ let emit_load_literal dst lbl =
     `	ldr	{emit_reg dst}, [{emit_reg reg_tmp1}, #:lo12:{emit_label lbl}]\n`
   end
 
-
-let move src dst =
-  if src.loc <> dst.loc then begin
-    match (src, dst) with
-    | {loc = Reg _; typ = Float}, {loc = Reg _} ->
-      `	fmov	{emit_reg dst}, {emit_reg src}\n`
-    | {loc = Reg _}, {loc = Reg _} ->
-      `	mov	{emit_reg dst}, {emit_reg src}\n`
-    | {loc = Reg _}, {loc = Stack _} ->
-      `	str	{emit_reg src}, {emit_stack dst}\n`
-    | {loc = Stack _}, {loc = Reg _} ->
-      `	ldr	{emit_reg dst}, {emit_stack src}\n`
-    | _ ->
-      assert false
-  end[@@warning "-4"]
+let move (src : Reg.t) (dst : Reg.t) =
+  let distinct = not (Reg.same_loc src dst) in
+  if distinct then
+  match src.typ, src.loc, dst.typ, dst.loc with
+  | Float, Reg _, Float, Reg _
+  | Float32, Reg _, Float32, Reg _
+  | (Vec128|Valx2), Reg _, (Vec128|Valx2), Reg _ ->
```

Comment:
Hmm, why has `Valx2` shown up already? 
Should this case be split into `Vec128, Vec128 | Valx2, Valx2` or do we allow moves between these types like with `Int/Val/Addr`?

Reply:
yes, it's the same for moves between `Int` and `Val`. `Vec128` and `Valx2` are treated the [same](https://github.com/ocaml-flambda/flambda-backend/blame/3bfc7a86bc18d8cab8e86bb83c0689be1485e3ba/backend/amd64/emit.ml#L829) in amd64 backend, and we had a similar discussion there when `Valx2` was added, can't find the comment now. IIRC register allocation can introduce these moves when coalescing.